src/ffi.rs

```diff
@@ -27,12 +37,36 @@ pub fn is_host_serial_interrupt(vector: u8) -> bool {
     unsafe { rvm_is_host_serial_interrupt(vector) }
 }
 
+#[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+pub fn riscv_trap_handler_no_frame(sepc: usize) {
+    let mut sepc_new = sepc;
+    unsafe { rvm_riscv_trap_handler_no_frame(&mut sepc_new) }
+    if sepc_new != sepc {
+        panic!("user_fixup called in trap handler, which should not happen for a vm trap.");
+    }
+}
+#[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+pub fn riscv_check_hypervisor_extension() -> bool {
+    unsafe { rvm_riscv_check_hypervisor_extension() }
+}
+
 extern "Rust" {
     fn rvm_alloc_frame() -> Option<HostPhysAddr>;
     fn rvm_dealloc_frame(_paddr: HostPhysAddr);
+    #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+    fn rvm_alloc_frame_x4() -> Option<HostPhysAddr>;
```

Comment:
It better to extend the `rvm_alloc_frame()` API to support allocate 4 frames, like `rvm_alloc_frame(frames: usize)`.